src/asdf/install.sh

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
+
+echo "Activating asdf ..."
+
+if [ "$(id -u)" -ne 0 ]; then
+    echo -e 'Script must be run as root. Use sudo, su, or add "USER root" to your Dockerfile before running this script.'
+    exit 1
+fi
+
+ASDF_VERSION=${VERSION:-"latest"}
+USER=${2:-"automatic"}
+
+export DEBIAN_FRONTEND=noninteractive
+
+# shellcheck source=/dev/null
+source /etc/os-release
+
+# Clean up
+cleanup() {
+case "${ID}" in
+    debian|ubuntu)
+      rm -rf /var/lib/apt/lists/*
+    ;;
+  esac
+}
+
+apt_get_update() {
+  case "${ID}" in
+    debian|ubuntu)
+      if [ "$(find /var/lib/apt/lists/* | wc -l)" = "0" ]; then
+        echo "Running apt-get update..."
+        apt-get update -y
+      fi
+    ;;
+  esac
+}
+
+check_git() {
+    if [ ! -x "$(command -v git)" ]; then
+        check_packages git
+    fi
+}
+
+# Checks if packages are installed and installs them if not
+check_packages() {
+  case "${ID}" in
+    debian|ubuntu)
+      if ! dpkg -s "$@" >/dev/null 2>&1; then
+        apt_get_update
+        apt-get -y install --no-install-recommends "$@"
+      fi
+    ;;
+    alpine)
+      if ! apk -e info "$@" >/dev/null 2>&1; then
+        apk add --no-cache "$@"
+      fi
+    ;;
+  esac
+}
+
+find_version_from_git_tags() {
```

Comment:
what is this?! :zombie:

Reply:
I know! It's a big function to get the latest version of a GitHub repository. If you have an idea to make it more simple.